### PR TITLE
Events v1

### PR DESCRIPTION
**Pull request by cameronn**
Campaign Monitor is releasing it's new behavioral data for Marketers capability in the near future. As a part of that, there is a new API for events. This pull request includes the addition of a wrapper to call those API methods for PHP.

### CODE REVIEW

File: csrest_events.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        $auth_details,
+        $client_id,
+        $protocol = 'https',
+        $debug_level = CS_REST_LOG_NONE,
+        $host = 'api.createsend.com',
+        $log = NULL,
+        $serialiser = NULL,
+        $transport = NULL) {
+            parent::__construct($auth_details, $protocol, $debug_level, $host, $log, $serialiser, $transport);
+            $this->set_client_id($client_id);
```
Comment (ishworgurung) [inline csrest_events.php]: if we don't validate `$client_id`, `$this->_events_base_route` is going to be invalid. https://secure.php.net/manual/en/function.isset.php is one possible solution.

File: csrest_events.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+
+
+        /**
+         * Change the client id used for calls after construction
+         * @param $client_id
+         * @access public
+         */
+        function set_client_id($client_id) {
+            $this->_events_base_route = $this->_base_route.'events/'.$client_id.'/';
```
Comment (ishworgurung) [inline csrest_events.php]: if we don't validate `$client_id`, `$this->_events_base_route` is going to be invalid. https://secure.php.net/manual/en/function.isset.php is one possible solution.

File: csrest_events.php (hunk 1 above)
Comment (cameronn) [inline csrest_events.php]: Updated

**Comment by ishworgurung** [comment]
LGTM

File: csrest_events.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+         *          array(
+         *              'RandomFieldObject''  => array(
+         *                                      'Example'' => 'test'
+         *                                  ),
+         *              'RandomFieldURL' => 'Example',
+         *              'RandomArray' => array(1,3,5,6,7),
+         *          )
+         *
+         * @access public
+         * @return CS_REST_Wrapper_Result A successful response will include a Event ID.
```
Comment (nufyoot) [inline csrest_events.php]: `include an Event ID`, change `a` to `an`?

File: csrest_events.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+        /**
+         * Change the client id used for calls after construction
+         * @param $client_id
+         * @access public
+         */
+        function set_client_id($client_id) {
+             if (!isset($client_id)) {
+                trigger_error('$client_id needs to be set');
+			    exit;
```
Comment (nufyoot) [inline csrest_events.php]: Would this terminate the script, including the PHP code that might be run to deliver a web page to an end user? Would this be desired behavior from a consumer view?

File: csrest_events.php [hunk 5]
```diff
@@ -0,0 +1,115 @@
+<?php
+require_once dirname(__FILE__).'/class/base_classes.php';
+
+/**
+ * Class to send event data to the create send API.
+ * @author philoye
```
Comment (cameronn) [inline csrest_events.php]: Done

File: csrest_events.php (hunk 4 above)
Comment (cameronn) [inline csrest_events.php]: done

File: csrest_events.php (hunk 3 above)
Comment (cameronn) [inline csrest_events.php]: done

File: csrest_events.php [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            }
+            if (!isset($event_name)) {
+                trigger_error('$event_name needs to be set');
+                return new CS_REST_Wrapper_Result(null, 400);
+            }
+            if (strlen($event_name) > 1000 ) {
+                trigger_error('$event_name needs to be shorter, max length is 1000 character');
+                return new CS_REST_Wrapper_Result(null, 400);
+            }   
+            if isset($data) {
```
Comment (gwigz) [inline csrest_events.php]: I'm guessing this `isset` is meant to have brackets around it, was this tested?

Comment (cameronn) [inline csrest_events.php]: Slightly embarrassed that I committed this typo, thanks for catching it.

**Pull request merged**